CoursesManager.UI/ViewModels/Students/CourseDialogViewModel.cs

```diff
@@ -0,0 +1,226 @@
+        {
+            _courseRepository = courseRepository ?? throw new ArgumentNullException(nameof(courseRepository));
+            _dialogService = dialogService ?? throw new ArgumentNullException(nameof(dialogService));
+            _locationRepository = locationRepository ?? throw new ArgumentNullException(nameof(locationRepository));
+
+            Course = course != null
+                   ? course.Copy()
+                    : new Course
+                    {
+                        Name = string.Empty,
+                        Code = string.Empty,
+                        Description = string.Empty,
+                        Location = null,
+                        IsActive = false,
+                        StartDate = DateTime.Now,
+                        EndDate = DateTime.Now,
+                        Image = null,
+
+                    };
+
+            OriginalCourse = course;
+            Locations = new ObservableCollection<Location>(locationRepository.GetAll());
+
+            Courses = new ObservableCollection<string>(_courseRepository.GetAll().Select(c => c.Name));
+            SaveCommand = new RelayCommand(ExecuteSave, () => CanSave);
+            CancelCommand = new RelayCommand(OnCancel);
+            UploadCommand = new RelayCommand(UploadImage);
+
+            Course.PropertyChanged += (_, _) => UpdateCanSave();
+            UpdateCanSave();
+        }
+
+        private Course? OriginalCourse { get; }
+        protected override void InvokeResponseCallback(DialogResult<Course> dialogResult)
+        {
+            ResponseCallback.Invoke(dialogResult);
+        }
+
+        private void NotifyCanExecuteChanged()
+        {
+            (SaveCommand as RelayCommand)?.RaiseCanExecuteChanged();
+        }
+
+        private async void ExecuteSave()
+        {
+            try
+            {
+                await OnSaveAsync();
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Error: {ex.Message}");
+            }
+        }
+
+        private void UpdateCanSave()
+        {
+            var fieldsAreValid = !string.IsNullOrWhiteSpace(Course!.Name)
+                                 && !string.IsNullOrWhiteSpace(Course.Code)
+                                 && Course.StartDate != default
+                                 && Course.EndDate != default
+                                 && Course.Location != null
+                                 && !string.IsNullOrWhiteSpace(Course.Description);
+
+            Console.WriteLine($"Fields Valid: {fieldsAreValid}");
+
+            var noValidationErrors = !HasValidationErrors();
+
+            Console.WriteLine($"No Validation Errors: {noValidationErrors}");
+
+            CanSave = fieldsAreValid && noValidationErrors;
+
+            NotifyCanExecuteChanged();
+        }
```

Comment:
Dit hele verhaal kan ook zonder de CanSave field. Beetje zoiets als dit

```csharp
private bool CanSave()
{
    return !string.IsNullOrWhiteSpace(Course?.Name) &&
           !string.IsNullOrWhiteSpace(Course?.Code) &&
           Course?.StartDate != default &&
           Course?.EndDate != default &&
           Course?.Location != null &&
           !string.IsNullOrWhiteSpace(Course?.Description) &&
           !HasValidationErrors();
}
```

Dan kun je die gewoon rechtstreeks toevoegen in de new RelayCommand(OnSave, CanSave)